### PR TITLE
[DOCS] Added settings page for ILM.

### PR DESCRIPTION
**Pull request by debadair**

### CODE REVIEW

File: docs/reference/settings/ilm-settings.asciidoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+templates. For more information, see <<ilm-gs-create-policy>>.
+
+`index.lifecycle.name`::
+The name of the policy to use to manage the index.
+
+`index.lifecycle.rollover_alias`::
+The index alias to update when the index rolls over. Required if the specified
+policy contains a rollover action. When the index rolls over, the alias is
+updated to reflect that the index is no longer the
+<<aliases-write-index, write index>>.
```
Comment (talevy) [inline docs/reference/settings/ilm-settings.asciidoc]: the subtlety here is that rolling-over with an explicit write index is not a requirement in the context of ILM. It is just highly recommended. Highly recommended in all cases, even outside of ILM 😄 . For this reason, should we link to the rollover docs here?

**Comment by gwbrown** [comment]
There's actually another setting, `index.lifecycle.indexing_complete`, described in the docs [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/skipping-rollover.html). It's mostly used internally though, and should probably be marked as an expert setting if we include it on this page.

File: docs/reference/index-modules.asciidoc [hunk 2]
```diff
@@ -256,6 +256,10 @@ Other index settings are available in index modules:
     Settings to define analyzers, tokenizers, token filters and character
     filters.
 
+[xpack]#<<ilm-settings,{ilm-cap}>>#::
```
Comment (lcawl) [inline docs/reference/index-modules.asciidoc]: The role attribute here generates

`<dt><span class="term"><span class="xpack xpack"><a class="link" href="ilm-settings.html" title="Index lifecycle management settings">Index lifecycle management</a></span></span></dt>`

... but we don't have anything for that in the https://github.com/elastic/docs/blob/master/resources/web/styles.css so  it doesn't have any effect.

**Review by lcawl — APPROVED** [review]
LGTM and builds successfully

File: docs/reference/index-modules.asciidoc (hunk 2 above)
Comment (debadair) [reply docs/reference/index-modules.asciidoc]: I moved it to a separate floated section so I could apply the attribute. I opened https://github.com/elastic/docs/issues/795 as a possible enhancement.

**Comment by debadair** [comment]
The main intent was to give more visibility to the settings users need to know about. I'm inclined to leave index.lifecycle.indexing_complete out of it, at least for now.

**Pull request merged**